core/math/a_star.cpp

```diff
@@ -259,8 +259,6 @@ int64_t AStar3D::get_point_capacity() const {
 }
 
 void AStar3D::reserve_space(int64_t p_num_nodes) {
-	ERR_FAIL_COND_MSG(p_num_nodes <= 0, vformat("New capacity must be greater than 0, new was: %d.", p_num_nodes));
```

Comment:
This should likely be kept, as this condition would still trigger an error in the template, and that's not a clear relationship to the caller of `AStar3D::reserve_space` that it's calling `reserve` on a container and how to interpret this error.

Reply:
Agreed!